### PR TITLE
[breaking change (new config option)] consolidate chain service configuration options 

### PR DESCRIPTION
**Pull request by snario**
~The `ChainService` should not be instantiated inside the `server-wallet` as it is fundamentally an external component that is designed to be provided _to_ the `server-wallet`. Similarly, the implementation details of the wallet should not need to know about mocks or test fixtures, that should be left to the tests. So, this PR moves the instantiation of a `ChainService` into the constructor of a `server-wallet`. Applications are expected to provide it in the constructor.~

~It seems reasonable to force the `ChainService` to be in the constructor as without it the `server-wallet` is not able to do things like make deposits, withdrawals, or monitor the chain. These are fundamental requirements of a state channels wallet.~

Simplified this PR to simply consolidate wallet config options into one place as they relate to the chain service after some discussions

### CODE REVIEW

File: packages/server-wallet/deployment/deploy.ts
Comment (kerzhner) [inline packages/server-wallet/deployment/deploy.ts:17]: https://github.com/statechannels/statechannels/blob/master/packages/devtools/src/constants.ts#L4

File: packages/server-wallet/jest/chain-setup.ts
Comment (kerzhner) [inline packages/server-wallet/jest/chain-setup.ts:15]: Have you tried changing these magic constants to ensure that this is the only place that relies these? Specifically, other places our code might rely on `process.env.GANACHE_PORT` and might default to port 8545. If that were the case, then changing 8545 to 8546 would result in failing tests.

File: packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts
Comment (kerzhner) [inline packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts:15]: This seems like a bad proliferation of a magic constant.

File: packages/server-wallet/src/chain-service/__chain-test__/chain-service.test.ts
Comment (kerzhner) [inline packages/server-wallet/src/chain-service/__chain-test__/chain-service.test.ts:28]: This needs to match the host and port of test setup phase. We should use either use environment variables or some other methodology.

**Review by kerzhner — APPROVED** [review]
A chain service defined externally seems like the right pattern to adopt. It would be nice to clean up the magic constants before this work is merged in.

**Review by andrewgordstewart — CHANGES_REQUESTED** [review]
I would like to temporarily block merging of this PR, not because I disagree with it (I agree with the change in theory)

The reason to block it is, this is actually a breaking change, not a refactor. Considering that, the managers will now need to be updated, so if
1. this gets into `master`
2. we want to fix some other issues, and release a new `server-wallet`
then we will be forced to also release this.

I'd like anyone (@snario ?) who has the time (I do not) to spend a few minutes thinking about if this poses an issue. If not, dismiss this review.

File: packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts
Comment (snario) [reply packages/server-wallet/src/__chain-test__/directly-funded-channel.test.ts:15]: Right now on `master` the jest tests in the `server-wallet` that use a chain all hard code the usage of `http://localhost:8545` inside `chain-setup.ts`. I think this change just makes it much more obvious. I think we may want to open a separate issue if we want to refactor the tests to easily have configurable chain endpoints.

**Pull request merged**
